tools/toolchain/image

```diff
@@ -1 +1 @@
-docker.io/scylladb/scylla-toolchain:fedora-38-20240321
+docker.io/scylladb/scylla-toolchain:fedora-38-20240506
```

Comment:
@xtrey You can't merge this PR with this change, it only contains the x86 docker image, we need to have a multi-arch update, and only then can you integrate this. Otherwise, we will break all arm-based building